addons/mail/models/mail_template.py

```diff
@@ -350,8 +376,12 @@ def render_template(self, template_txt, model, res_ids, post_process=False):
             mako_env = mako_safe_template_env if self.env.context.get('safe') else mako_template_env
             template = mako_env.from_string(tools.ustr(template_txt))
         except Exception:
-            _logger.info("Failed to load template %r", template_txt, exc_info=True)
-            return multi_mode and results or results[res_ids[0]]
+            _logger.warn("Failed to load template:\n%r",
+                         template_txt, exc_info=True)
+            raise UserError(_(
```

Comment:
Why changing the return to an exception ?

Reply:
If the user sends the template on a wizard, without saving, the rendering might still crash but the mail will be sent anyways (empty mail).

I was actually thinking of raising the exception in the write override since that's all that's needed to block it when saving, but thought maybe we wanted the same for when sending directly without saving a template.

Reply:
btw, if you want me to keep the return and raise inside the write(), can I change the return to `return results if multi_mode else results[res_ids[0]]` ? More explicit that way